x/migration/module/cmd/claim_morse_supplier.go

```diff
@@ -0,0 +1,203 @@
+	clientCtx, err := cosmosclient.GetClientTxContext(cmd)
+	if err != nil {
+		return err
+	}
+
+	// Load the supplier stake config from the YAML file.
+	supplierStakeConfigPath := args[1]
+	supplierStakeConfig, err := loadSupplierStakeConfigYAML(supplierStakeConfigPath)
+	if err != nil {
+		return err
+	}
+
+	// Ensure that the signing account matches either the configured owner or operator address.
+	signingAddr := clientCtx.GetFromAddress().String()
+	ownerAddr := supplierStakeConfig.OwnerAddress
+	operatorAddr := supplierStakeConfig.OperatorAddress
+	switch signingAddr {
+	case ownerAddr, operatorAddr:
+		// All good.
+	default:
+		return fmt.Errorf(
+			"signer address %s does not match owner address %s or supplier operator address %s",
+			signingAddr, ownerAddr, operatorAddr,
+		)
+	}
+
+	// Construct a MsgClaimMorseSupplier message.
+	msgClaimMorseSupplier, err := types.NewMsgClaimMorseSupplier(
+		ownerAddr,
+		operatorAddr,
+		morsePrivKey.PubKey().Address().String(),
+		morsePrivKey,
+		supplierStakeConfig.Services,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Serialize, as JSON, and print the MsgClaimMorseSupplier for posterity and/or confirmation.
+	msgClaimMorseSupplierJSON, err := json.MarshalIndent(msgClaimMorseSupplier, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("MsgClaimMorseSupplier %s\n", string(msgClaimMorseSupplierJSON))
+
+	// Last chance for the user to abort.
+	skipConfirmation, err := cmd.Flags().GetBool(cosmosflags.FlagSkipConfirmation)
+	if err != nil {
+		return err
+	}
+
+	if !skipConfirmation {
+		fmt.Printf("Confirm MsgClaimMorseSupplier: y/[n]: ")
+		stdinReader := bufio.NewReader(os.Stdin)
+
+		// This call to ReadLine() will block until the user sends a new line to stdin.
+		inputLine, _, readErr := stdinReader.ReadLine()
+		if readErr != nil {
+			return err
+		}
+
+		// Terminate the confirmation prompt output line.
+		fmt.Println()
+
+		// Abort unless some affirmative confirmation is given.
+		switch string(inputLine) {
+		case "Yes", "yes", "Y", "y":
+		default:
+			return nil
+		}
+	}
+
+	// Construct a tx client.
+	txClient, err := flags.GetTxClient(ctx, cmd)
+	if err != nil {
+		return err
+	}
+
+	// Sign and broadcast the claim Morse account message.
+	eitherErr := txClient.SignAndBroadcast(ctx, msgClaimMorseSupplier)
+	err, errCh := eitherErr.SyncOrAsyncError()
+	if err != nil {
+		return err
+	}
+
+	// Wait for an async error, timeout, or the errCh to close on success.
+	return <-errCh
+}
+
+// loadSupplierStakeConfigYAML loads, parses, and validates the supplier stake
+// config from configYAMLPath.
+func loadSupplierStakeConfigYAML(configYAMLPath string) (*config.SupplierStakeConfig, error) {
+	yamlStakeConfigBz, err := os.ReadFile(configYAMLPath)
+	if err != nil {
+		return nil, err
+	}
+
+	var yamlStakeConfig config.YAMLStakeConfig
+	if err = yaml.Unmarshal(yamlStakeConfigBz, &yamlStakeConfig); err != nil {
+		return nil, err
+	}
+
+	if len(yamlStakeConfig.StakeAmount) != 0 {
```

Comment:
Thoughts on enforcing this (in the CLI) to be the same as what's onchain for validation purposes?